src/vcpkg/commands.ci.cpp

```diff
@@ -50,13 +50,24 @@ namespace
         {SwitchXXUnitAll, msgCISwitchOptXUnitAll},
     };
 
-    struct UnknownCIPortsResults
+    struct CIPreBuildStatus
```

Comment:
This type mixes logic results (known) with formatting results (report_lines / abis); it might be cleaner to reduce this to just `std::map<PackageSpec, BuildResult> known;` and have the report code deal with calculating report_lines / abis on the fly from the action_plan and/or excluded.

Request no change; consider follow-up PR.